vexim/config/functions.php

```diff
@@ -43,6 +43,24 @@ function check_user_exists($dbh,$localpart,$domain_id,$page)
         }
     }
 
+    /**
+     * Check if mail address is compliant with RFC 3696.
+     *
+     * Queries database $dbh, and redirects to the $page is the user already
+     * exists.
+     *
+     * @param  string  $localpart
+     * @param  string  $domain
+     * @param  string  $page       page to return to
+     */
+    function check_mail_address($localpart,$domain,$page)
+    {
+      if ((strlen($localpart)+strlen($domain)>253) || strlen($localpart)>64)
+        {
+            header ("Location: $page?addresstoolong=$localpart");
```

Comment:
This doesn't seem to account for the possibility of vexim being hosted in a subdirectory.